packages/contracts/src/chugsplash/deploy.ts

```diff
@@ -0,0 +1,62 @@
+/* Imports: External */
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+import { ethers, Signer } from 'ethers'
+import { sleep } from '@eth-optimism/core-utils'
+
+/* Imports: Internal */
+import { getContractFactory } from '../contract-defs'
+import { makeActionBundleFromConfig } from './hardhat-tools'
+import { ChugSplashConfig } from './config'
+
+const DEPLOYER_PRIVATE_KEY = process.env.DEPLOYER_PRIVATE_KEY
+const L2_NODE_URL = process.env.L2_NODE_URL
+const CHUGSPLASH_DEPLOYER_ADDRESS = process.env.CHUGSPLASH_DEPLOYER_ADDRESS
+const UPGRADE_CONFIG_PATH = process.env.UPGRADE_CONFIG_PATH
+
+export const executeActionsFromConfig = async (
+  hre: HardhatRuntimeEnvironment,
+  signer: Signer,
+  chugsplashDeployerAddress: string,
+  upgradeConfigPath: string
```

Comment:
Looking at this now I think it might be better to feed a `ChugSplashConfig` into here and have the hardhat task be responsible for converting from a path to a JSON object via `require`. I think it makes this a little easier to test.

Reply:
Would also allow you to get rid of the `example-configs` for now.

Reply:
^open to changing this, but i'm inclined to make the hardhart task as dumb as possible and have this function handle the errors. added a test around invalid config which seems good to catch